struct.c

```diff
@@ -915,7 +989,12 @@ static VALUE
 inspect_struct(VALUE s, VALUE dummy, int recur)
 {
     VALUE cname = rb_class_path(rb_obj_class(s));
-    VALUE members, str = rb_str_new2("#<struct ");
+    VALUE members, str;
+    if (rb_obj_is_kind_of(s, rb_cData) == Qtrue) {
+       str = rb_str_new2("#<data ");
+    } else {
+       str = rb_str_new2("#<struct ");
+    }
```

Comment:
I think it'd be better to pass the prefix from `rb_struct_inspect` via `dummy` argument.

```suggestion
    VALUE members, str = rb_str_new2((const char *)prefix);
```